src/core/metadata_manager.py

```diff
@@ -39,27 +39,26 @@
 from lib.timing import get_utc_now, global_start
 
 
-@Singleton
 class MetadataManager:
-    metadata_extractors: list[type(MetadataBase)] = []
-
-    blacklisted_for_cache = [MaliciousExtensions, ExtractFromFiles, Javascript]
-
     def __init__(self) -> None:
         self._logger = get_logger()
-        self._setup_extractors()
+        self.metadata_extractors: list[MetadataBase] = [
+            self._setup_extractor(extractor) for extractor in self._extractors()
+        ]
         self.tld_extractor: TLDExtract = TLDExtract(cache_dir=None)
+        self.blacklisted_for_cache = [MaliciousExtensions, ExtractFromFiles, Javascript]
+        self.cache_manager = CacheManager()
```

Comment:
So cache manager now belongs to the metadata manager and is nowhere else used, correct?

Reply:
Correct! Technically the cache manager is only a proxy to the underlying storage and having multiple proxies to the same storage. This means, even if there would be multiple CacheManager instances around the application, they should not interfere with each other?!